tools/omnissm/api/register/register.go

```diff
@@ -0,0 +1,289 @@
+	registration.ActivationID = *activateResult.ActivationId
+	registration.ManagedID = i.GetManagedID()
+
+	regRecord, err := dynamodbattribute.MarshalMap(registration)
+	insertParams := &dynamodb.PutItemInput{
+		Item:      regRecord,
+		TableName: aws.String(RegistrationTable),
+	}
+
+	insertRequest := dbClient.PutItemRequest(insertParams)
+	insertResult, err := insertRequest.Send()
+
+	if err != nil {
+		fmt.Println("Put Registration Error", insertResult, err)
+		return registration, err
+	}
+	return registration, nil
+}
+
+// InstanceRegistration Minimal
+type InstanceRegistration struct {
+	ActivationCode string
+	ActivationID   string `json:"ActivationId"`
+	ManagedID      string `json:"mid"`
+}
+
+func validateRequest(requestBody string) (InstanceIdentity, events.APIGatewayProxyResponse) {
+
+	var regRequest RegistrationRequest
+	var identity InstanceIdentity
+
+	err := json.Unmarshal([]byte(requestBody), &regRequest)
+	if err != nil {
+		response := map[string]string{
+			"error":   "invalid-request",
+			"message": "malformed json",
+		}
+		body, _ := json.Marshal(response)
+		return identity, events.APIGatewayProxyResponse{Body: string(body), StatusCode: 400}
+	}
+
+	// TODO: At the moment this is AWS Specific, support GCP & Azure to the extant possible.
+	switch regRequest.Provider {
+	case "aws":
+	default:
+		response := map[string]string{
+			"error":   "invalid-request",
+			"message": "unknown provider"}
+
+		body, _ := json.Marshal(response)
+		return identity, events.APIGatewayProxyResponse{Body: string(body), StatusCode: 400}
+	}
+
+	signature, err := base64.StdEncoding.DecodeString(string(regRequest.Signature))
+	if err != nil {
+		response := map[string]string{
+			"error":   "invalid-request",
+			"message": "malformed rsa signature",
+		}
+		body, _ := json.Marshal(response)
+		return identity, events.APIGatewayProxyResponse{Body: string(body), StatusCode: 400}
+	}
+	err = RSACert.CheckSignature(x509.SHA256WithRSA, []byte(regRequest.Identity), signature)
+	if err != nil {
+		response := map[string]string{
+			"error":   "invalid-signature",
+			"message": "invalid identity",
+		}
+		body, _ := json.Marshal(response)
+		return identity, events.APIGatewayProxyResponse{Body: string(body), StatusCode: 400}
+	}
+
+	// We verified the signature, so malformed here would more than odd.
+	_ = json.Unmarshal([]byte(regRequest.Identity), &identity)
+
+	return identity, events.APIGatewayProxyResponse{}
+}
+
+func handleRequest(ctx context.Context, request events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+
+	fmt.Printf("Processing request data for request %s.\n", request.RequestContext.RequestID)
+	fmt.Printf("Body size = %d.\n", len(request.Body))
+
+	identity, errorResponse := validateRequest(request.Body)
+
+	if len(identity.InstanceID) < 1 {
+		return errorResponse, nil
+	}
+
+	fmt.Println("Instance Registration Request",
+		identity.InstanceID, identity.Region, identity.AccountID, identity.GetManagedID())
```

Comment:
The most common style here is to not limit the length of lines.

Reply:
pep8ism, 80 char terminals 4ever, fixed.